src/createClassProxy.js

```diff
@@ -170,4 +171,21 @@ export default function proxyClass(InitialComponent) {
   });
 
   return proxy;
-}
\ No newline at end of file
+}
+
+function createFallback(Component) {
+  let CurrentComponent = Component;
+
+  return {
+    get() {
+      return CurrentComponent;
+    },
+    update(NextComponent) {
+      CurrentComponent = NextComponent;
+    }
+  };
+}
+
+export default function createClassProxy(Component) {
+  return supportsProtoAssignment(Component) ? proxyClass(Component) : createFallback(Component);
+}
```

Comment:
I originally had this check inside of `proxyClass`, but this was the cleanest way I could think of to prevent as much churn as possible.